### PR TITLE
fix: change describeStackResources() call to listStackResources()

### LINKED ISSUE #9037: Storage outputs from cfn not getting generated
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v12.22.1

### Amplify CLI Version

7.4.5

### What operating system are you using?

Mac

### Amplify Categories

auth, storage

### Amplify Commands

push

### Describe the bug

Entire environment is broken after trying to upgrade to amplify v7.4.5.

Issue started with our cypress tests failing with our front-end getting an error on Storage.put indicating that the region is missing. This led me down to realized that the s3/storage information is not getting populated into the aws-exports.json file.

I then noticed on every push it is saying that our storage needs to be created (s3), even though it definitely exists.

I then went through the process of doing the overrides migration https://docs.amplify.aws/cli/migration/override/ in the attempt to fix this. I started with storage and did a push which then failed. I then tried doing Auth which resulted in more failures.

So now I have reverted everything and went back to version 6.4.0 which is working.

The following was the latest push failure which means that during your migration process, something is not referencing the new outputs/inputs from auth.

I replaced our resource name with XXXXXX for privacy purposes.
```
UPDATE_FAILED storageXXXXXX   AWS::CloudFormation::Stack Mon Nov 22 2021 15:13:04 GMT-0800 (Pacific Standard Time) Parameters: [authXXXXXXXXXXXXXUserPoolId, authuserPoolGroupsadminGroupRole] do not exist in the template
UPDATE_FAILED apiXXXXXX AWS::CloudFormation::Stack Mon Nov 22 2021 15:13:05 GMT-0800 (Pacific Standard Time) Resource update cancelled   
```

### Expected behavior

amplify push works and amplify-exports.json is generated correctly

### Reproduction steps

See above description

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by kaustavghosh06**
#### Issue #, if available
fixes #9037

#### Description of how you validated changes

AWS.CloudFormation.DescribeStackResources() only returns 100 resources (https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_DescribeStackResources.html) due to it's limits and it is recommended to use ListStackResources() instead. This is causing projects with > 100 Amplify resources to fail to fetch outputs from the nested cloudformation stacks and populate it in the amplify-meta.json file.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by akshbhu — APPROVED** [review]

File: packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js [hunk 1]
```diff
@@ -326,9 +326,10 @@ class CloudFormation {
     };
     const projectDetails = this.context.amplify.getProjectDetails();
     const { amplifyMeta } = projectDetails;
-    logger('updateamplifyMetaFileWithStackOutputs.cfn.describeStackResources', [cfnParentStackParams])();
-    const result = await this.cfn.describeStackResources(cfnParentStackParams).promise();
-    const resources = result.StackResources.filter(
+    logger('updateamplifyMetaFileWithStackOutputs.cfn.listStackResources', [cfnParentStackParams])();
+    const result = await this.cfn.listStackResources(cfnParentStackParams).promise();
+    console.log(result);
```
Comment (jhockett) [inline packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js:331]: nit: remove console.log

Comment (jhockett) [reply packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js:331]:
```suggestion
```

Comment (kaustavghosh06) [reply packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js]: Done.

**Review by jhockett — APPROVED** [review]

**Comment by tomhirschfeld** [comment]
@jhockett @kaustavghosh06 any chance we can get a release with this in it tomorrow? This issue is blocking our production deploys and local development

**Comment by kaustavghosh06** [comment]
@tomhirschfeld Yes, we're working on a release for this issue with urgency right now.

**Comment by tomhirschfeld** [comment]
thanks! I appreciate it! Feeling very taken care of with the resolution of this issue!

**Comment by gakinson** [comment]
Testing it now

**Comment by gakinson** [comment]
Hate to say this, but still facing the same issue, are you sure you got all the correct calls to be switched?

**Comment by kaustavghosh06** [comment]
@gakinson Yes, the one which populates the amplify-meta configurations, we have that switched. Could you please confirm the amplify cli version is 7.5.1 that you're currently using?

**Comment by gakinson** [comment]
i did amplify --version which is showing 7.5.1

trying to do a force push to see if that works

**Comment by gakinson** [comment]
Seperate error.

I am still trying locally, but I thought I would try upgrading the CI/CD pipline which deploys to our beta environment. Had the following error:

```
2021-11-24T07:32:29.692Z [INFO]: 
2021-11-24T07:32:40.141Z [INFO]: [33mNote: It is recommended to run this command from the root of your app directory[39m
2021-11-24T07:32:40.799Z [WARNING]: - Initializing your environment: master
2021-11-24T07:33:20.026Z [WARNING]: - Building resource auth/wesponsoredweb1966eb90
2021-11-24T07:33:20.028Z [WARNING]: - Building resource auth/userPoolGroups
2021-11-24T07:33:20.114Z [WARNING]: - Building resource api/wesponsoredweb
2021-11-24T07:33:20.115Z [WARNING]: - Building resource api/stripeBridge
2021-11-24T07:33:20.115Z [WARNING]: - Building resource api/socialBridge
2021-11-24T07:33:20.116Z [WARNING]: - Building resource storage/invoices
2021-11-24T07:33:20.116Z [WARNING]: - Building resource storage/PostIngestImportHistory
2021-11-24T07:33:20.116Z [WARNING]: - Building resource storage/ReviewIngestHistory
2021-11-24T07:33:20.117Z [WARNING]: - Building resource storage/BudgetExpenditure
2021-11-24T07:33:20.117Z [WARNING]: - Building resource storage/SocialVerificationCodes
2021-11-24T07:33:20.136Z [WARNING]: ✔ Initialized provider successfully.
2021-11-24T07:33:20.166Z [WARNING]: - Retrieving channel information for APNS.
2021-11-24T07:33:20.306Z [WARNING]: ✔ Channel information retrieved for APNS
2021-11-24T07:33:20.306Z [WARNING]: - Retrieving channel information for FCM.
2021-11-24T07:33:20.468Z [WARNING]: ✔ Channel is not setup for FCM
2021-11-24T07:33:20.469Z [WARNING]: - Retrieving channel information for Email.
2021-11-24T07:33:20.676Z [WARNING]: ✔ Channel information retrieved for Email
2021-11-24T07:33:20.677Z [WARNING]: - Retrieving channel information for SMS.
2021-11-24T07:33:20.871Z [WARNING]: ✔ Channel information retrieved for SMS
2021-11-24T07:33:29.731Z [WARNING]: - Uploading files...
2021-11-24T07:33:36.660Z [INFO]: [31mUnable to import custom transformer module(graphql-ttl-transformer).[39m
                                 [31mYou may fix this error by editing transformers at /codebuild/output/src399252330/src/wesponsored-web/amplify/backend/api/wesponsoredweb/transform.conf.json[39m
2021-11-24T07:33:36.661Z [WARNING]: ✖ An error occurred when pushing the resources to the cloud
2021-11-24T07:33:36.730Z [WARNING]: ✖ There was an error initializing your environment.
2021-11-24T07:33:36.734Z [INFO]: [0m/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/import-global/index.js:1[0m
                                 [0mError: Cannot find module '/root/.nvm/versions/node/v12.21.0/lib/node_modules/graphql-ttl-transformer'[0m
                                 [0mRequire stack:[0m
                                 [0m- /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/import-global/index.js[0m
                                 [0m- /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/lib/transform-graphql-schema.js[0m
                                 [0m- /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/lib/push-resources.js[0m
                                 [0m- /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/lib/override-manager/transform-resource.js[0m
                                 [0m- /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/lib/override-manager/index.js[0m
                                 [0m- /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/lib/initializer.js[0m
                                 [0m- /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/lib/index.js[0m
                                 [0m- /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/lib/extensions/amplify-helpers/execute-provider-utils.js[0m
                                 [0m- /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/lib/domain/amplify-toolkit.js[0m
                                 [0m- /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/lib/domain/context.js[0m
                                 [0m- /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/lib/context-manager.js[0m
                                 [0m- /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/lib/index.js[0m
                                 [0m- /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/bin/amplify[0m
                                 [0m    at module.exports (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/import-global/index.js:9:10)[0m
                                 [0m    at /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/transform-graphql-schema.ts:141:44[0m
                                 [0m    at Array.map (<anonymous>)[0m
                                 [0m    at Object.transformersFactory (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/transform-graphql-schema.ts:110:8)[0m
                                 [0m    at _buildProject (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/graphql-transformer-core/src/util/amplifyUtils.ts:85:24)[0m
                                 [0m    at ensureMissingStackMappings (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/graphql-transformer-core/src/util/amplifyUtils.ts:166:23)[0m
                                 [0m    at buildProject (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/graphql-transformer-core/src/util/amplifyUtils.ts:51:3)[0m
                                 [0m    at transformGraphQLSchema (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/transform-graphql-schema.ts:526:29)[0m
                                 [0m    at Object.run (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/push-resources.ts:164:5) {[0m
                                 [0m  code: 'MODULE_NOT_FOUND',[0m
                                 [0m  requireStack: [[0m
                                 [0m    '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/import-global/index.js',[0m
                                 [0m    '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/lib/transform-graphql-schema.js',[0m
                                 [0m    '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/lib/push-resources.js',[0m
                                 [0m    '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/lib/override-manager/transform-resource.js',[0m
                                 [0m    '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/lib/override-manager/index.js',[0m
                                 [0m    '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/lib/initializer.js',[0m
                                 [0m    '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/lib/index.js',[0m
                                 [0m    '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/lib/extensions/amplify-helpers/execute-provider-utils.js',[0m
                                 [0m    '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/lib/domain/amplify-toolkit.js',[0m
                                 [0m    '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/lib/domain/context.js',[0m
                                 [0m    '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/lib/context-manager.js',[0m
                                 [0m    '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/lib/index.js',[0m
                                 [0m    '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/bin/amplify'[0m
                                 [0m  ][0m
                                 [0m}[0m
2021-11-24T07:33:36.778Z [ERROR]: !!! Build failed
2021-11-24T07:33:36.778Z [ERROR]: !!! Non-Zero Exit Code detected
2021-11-24T07:33:36.778Z [INFO]: # Starting environment caching...
2021-11-24T07:33:36.779Z [INFO]: # Uploading environment cache artifact...
2021-11-24T07:33:37.046Z [INFO]: # Environment caching completed
Terminating logging...
```

**Comment by kaustavghosh06** [comment]
The above error looks unrelated - but it might be tied to not having the custom transformer installed (graphql-ttl-transformer) as a part of the build spec and not referencing it in the transform.conf.json file as mentioned here - https://docs.amplify.aws/cli/plugins/authoring/#adding-custom-graphql-transformers-to-the-project

**Comment by gakinson** [comment]
> The above error looks unrelated - but it might be tied to not having the custom transformer installed (graphql-ttl-transformer) as a part of the build spec and not referencing it in the transform.conf.json file as mentioned here - https://docs.amplify.aws/cli/plugins/authoring/#adding-custom-graphql-transformers-to-the-project

Did something recently change for this to happen, it never came up before?

Do you know what exactly we need to add to the buildspec?

**Comment by gakinson** [comment]
> The above error looks unrelated - but it might be tied to not having the custom transformer installed (graphql-ttl-transformer) as a part of the build spec and not referencing it in the transform.conf.json file as mentioned here - https://docs.amplify.aws/cli/plugins/authoring/#adding-custom-graphql-transformers-to-the-project

Figured out how to resolve this one.

Still waiting on force push to complete

**Comment by gakinson** [comment]
Still not generating the correct amplify-meta or aws-exports.json locally

```
"storage": {
    "invoices": {
      "service": "S3",
      "providerPlugin": "awscloudformation",
      "dependsOn": [
        {
          "category": "auth",
          "resourceName": "wesponsoredweb1966eb90",
          "attributes": [
            "UserPoolId"
          ]
        },
        {
          "category": "auth",
          "resourceName": "userPoolGroups",
          "attributes": [
            "adminGroupRole"
          ]
        }
      ],
      "providerMetadata": {
        "s3TemplateURL": "https://s3.amazonaws.com/amplify-wesponsored-web-gakinson-173449-deployment/amplify-cfn-templates/storage/s3-cloudformation-template.json",
        "logicalId": "storageinvoices"
      },
      "lastPushTimeStamp": "2021-11-24T07:46:44.134Z",
      "lastPushDirHash": "8kID/Go1yO+uakqvDQFfMOCRek4="
    },
```

**Comment by kaustavghosh06** [comment]
I'm cloning a replica of your environment to repro locally with 100+ resources for debugging. The resources are pushing to the cloud at the moment.

**Comment by gakinson** [comment]
okay awesome. I am going to go to bed for the night, but will sync in the morning.

I am creating a brand new environment as well to see if that works.

**Comment by tomhirschfeld** [comment]
One thing I noticed looking at the new cfn api call and this code change is that this new call is Paginated but the way it's implemented doesn't use page tokens to call for all resources. Haven't tested it but might an issue

**Comment by gakinson** [comment]
https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_ListStackResources.html

https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/CloudFormation.html#listStackResources-property

Looking here, there is the concept of next tokens for list stack resources if the response is greater than 1MB

**Comment by kaustavghosh06** [comment]
@gakinson Yeah, I checked the limit out last night, but the 1 MB limit seems reasonable to fit in 100 records. I wasn't able to clone your environment locally, but I have some ideas to debug this locally with you and sent you an email requesting for a time.

**Comment by gakinson** [comment]
Replied to your email.

So i just tested deleting resources so that we had 100 total resources and the output is being generated. I think this is still related to this describeStackResources vs listStackResources.

**Comment by gakinson** [comment]
Tested with my code changes on the CLI and it seemed to have work. It definitely seems like the pagination thing.

This was the code I wrote, please do not use it exactly as it is as it is fairly messy as I was just trying to quickly test this:


```
        const cfnParentStackParams = {
                StackName: parentStackName
        };
        const projectDetails = this.context.amplify.getProjectDetails();
        const { amplifyMeta } = projectDetails;
        let NextToken
        let stackSummaries = []
        do {
            const listStackResourcesParams = {
                StackName: parentStackName,
                NextToken
            };
            logger('updateamplifyMetaFileWithStackOutputs.cfn.listStackResources', [listStackResourcesParams])();

            const result = await this.cfn.listStackResources(listStackResourcesParams).promise();
            console.log(result)
            NextToken = result.NextToken
            stackSummaries.push(...result.StackResourceSummaries)
        } while(NextToken)
        const resources = stackSummaries.filter(resource => ![
            'DeploymentBucket',
            'AuthRole',
            'UnauthRole',
            'UpdateRolesWithIDPFunction',
            'UpdateRolesWithIDPFunctionOutputs',
            'UpdateRolesWithIDPFunctionRole',
        ].includes(resource.LogicalResourceId) && resource.ResourceType === 'AWS::CloudFormation::Stack');
        const rootStackResources = stackSummaries.filter(resource => !['UpdateRolesWithIDPFunction', 'UpdateRolesWithIDPFunctionOutputs', 'UpdateRolesWithIDPFunctionRole'].includes(resource.LogicalResourceId));
        
```

**Comment by kaustavghosh06** [comment]
@gakinson - https://github.com/aws-amplify/amplify-cli/pull/9073/files

**Pull request merged**